### PR TITLE
Added Sleepsort [golang]

### LINKED ISSUE #427: Sleep Sort [Golang]
#### This issue is about [Algo/DS Name](link to resource for the Algo/DS).

- [x] Issue Title is in form `Algo/DS Name [Language]`.
- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [x] This Algo/DS is worth adding to this repository.
        - it uses channels
- [x] Assign this issue to me (I want to work on this).
	- [x] I ensure that I am not already assigned to 2 or more issues.
- [ ] I will send my PR only after I'm assigned to this issue by any maintainer.
         - i failed: https://github.com/iiitv/algos/pull/425
- [x] I understand that I will be unassigned from this issue if I fail to send a PR for this within a week.

### PR DESCRIPTION
**Pull request by hloeffler**
Fixes #427

#### By submitting this pull request I confirm I've read and complied with the below declarations.

- [ ] I have read the [Contribution guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [coding guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#cs) for this project.
- [x] My code follows the [skeleton code structure](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#sample).
- [x] This pull request has a descriptive title. For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [x] This pull request will be closed if I fail to update it even once in a continuous time span of 7 days.

### CODE REVIEW

**Comment by mohitkyadav** [comment]
@hloeffler Please read the guidelines. Create an issue before sending a PR.
Read https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md

File: sleep_sort/sleep_sort.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import (
+	"fmt"
+	"time"
+)
+
+// SleepSort sorts an array using SleepSort algorithm
+func SleepSort(numbers []int) []int {
+
+	channel := make(chan int, len(numbers))
+
```
Comment (aviaryan) [inline sleep_sort/sleep_sort.go]: Hey, @hloeffler. Thanks for the PR. Everything looks good. 
Just one thing, I see a lot of empty lines mixed with the code. Do you see any advantage to this? 
The tests are passing so it won't be right to enforce removing these lines but if you agree with the same, please go ahead and do so.
Thanks

**Comment by hloeffler** [comment]
sure, will do

**Comment by hloeffler** [comment]
@aviaryan better?

**Pull request merged**